### ISSUE TITLE
prova ml

### ISSUE THREAD

**Issue by thomasalves01** [thomasalves01/prova#1]
import pandas as pd

# Substitua 'seu_dataset.csv' pelo caminho do seu arquivo CSV
df = pd.read_csv('prova08_05.csv')

valores_ausentes = df.isnull().sum()
print("Valores Ausentes por Coluna:")
print(valores_ausentes)

duplicados = df[df.duplicated()]
print("Linhas Duplicadas:")
print(duplicados)

# Exibir os primeiros registros do DataFrame
print("Primeiros Registros:")
print(df.head())

import matplotlib.pyplot as plt

# Plotar um histograma de uma coluna específica (substitua 'nome_da_coluna' pelo nome real da coluna)
plt.hist(df['Preço_do_Produto'], bins=20)  # Você pode ajustar o número de bins conforme necessário
plt.xlabel('valores')
plt.ylabel('Frequência')
plt.title('Quantidade_Vendida')
plt.show()

# Plotar um boxplot de uma coluna específica (substitua 'nome_da_coluna' pelo nome real da coluna)
plt.boxplot(df['ID_do_Produto'])
plt.ylabel('Valores')
plt.title('Boxplot de Valores')
plt.show()

# Plotar um gráfico de dispersão de duas colunas (substitua 'coluna_x' e 'coluna_y' pelos nomes reais das colunas)
plt.scatter(df['ID_do_Produto'], df['Data_e_Hora'])
plt.xlabel('Coluna X')
plt.ylabel('Coluna Y')
plt.title('Gráfico de Dispersão')
plt.show()

# Remover colunas não numéricas (se necessário)
df_numeric = df.select_dtypes(include=['number'])

# Calcular estatísticas básicas apenas nas colunas numéricas
if not df_numeric.empty:
    media = df_numeric.mean()
    mediana = df_numeric.median()
    desvio_padrao = df_numeric.std()

    # Exibir as estatísticas
    print("Média:")
    print(media)
    print("\nMediana:")
    print(mediana)
    print("\nDesvio Padrão:")
    print(desvio_padrao)
else:
    print("Não há colunas numéricas no conjunto de dados.")

import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
import matplotlib.pyplot as plt

# Gerar dados de exemplo
np.random.seed(0)
X = 2 * np.random.rand(100, 1)  # Variável independente
y = 4 + 3 * X + np.random.randn(100, 1)  # Variável dependente

# Dividir os dados em conjunto de treinamento e conjunto de teste
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Criar e treinar o modelo de regressão linear
modelo = LinearRegression()
modelo.fit(X_train, y_train)

# Coeficientes do modelo
coeficiente_angular = modelo.coef_[0][0]
intercepto = modelo.intercept_[0]
print("Coeficiente Angular:", coeficiente_angular)
print("Intercepto:", intercepto)

# Fazer previsões no conjunto de teste
y_pred = modelo.predict(X_test)

# Visualizar os resultados
plt.scatter(X_test, y_test, color='blue')
plt.plot(X_test, y_pred, color='red')
plt.xlabel('X')
plt.ylabel('Y')
plt.title('Regressão Linear Simples')
plt.show()

# Calcular os resíduos
residuos = y_test - y_pred

# Plotar o gráfico de dispersão dos resíduos
plt.scatter(y_pred, residuos, color='green')
plt.axhline(y=0, color='red', linestyle='--')  # Adicionar linha horizontal em y=0
plt.xlabel('Valores Previstos')
plt.ylabel('Resíduos')
plt.title('Gráfico de Dispersão dos Resíduos')
plt.show()
